app/assets/javascripts/solidus_stripe/stripe-init/base.js

```diff
@@ -0,0 +1,173 @@
+Spree.stripePaymentMethod = {
+  config: $('[data-stripe-config').data('stripe-config'),
+  requestShipping: false
+}
+
+var authToken = $('meta[name="csrf-token"]').attr('content');
+
+var stripe = Stripe(Spree.stripePaymentMethod.config.publishable_key)
+var elements = stripe.elements({locale: 'en'});
+
+var element = $('#payment_method_' + Spree.stripePaymentMethod.config.id);
+var form = element.parents('form');
+var errorElement = form.find('#card-errors');
+var submitButton = form.find('input[type="submit"]');
+
+function stripeTokenHandler(token) {
+  var baseSelector = `<input type='hidden' class='stripeToken' name='payment_source[${Spree.stripePaymentMethod.config.id}]`;
+
+  element.append(`${baseSelector}[gateway_payment_profile_id]' value='${token.id}'/>`);
+  element.append(`${baseSelector}[last_digits]' value='${token.card.last4}'/>`);
+  element.append(`${baseSelector}[month]' value='${token.card.exp_month}'/>`);
+  element.append(`${baseSelector}[year]' value='${token.card.exp_year}'/>`);
+  form.find('input#cc_type').val(mapCC(token.card.brand || token.card.type));
+};
+
+function initElements() {
+  var style = {
+    base: {
+      color: 'black',
+      fontFamily: '"Helvetica Neue", Helvetica, sans-serif',
+      fontSmoothing: 'antialiased',
+      fontSize: '14px',
+      '::placeholder': {
+        color: 'silver'
+      }
+    },
+    invalid: {
+      color: 'red',
+      iconColor: 'red'
+    }
+  };
+
+  elements.create('cardExpiry', {style: style}).mount('#card_expiry');
+  elements.create('cardCvc', {style: style}).mount('#card_cvc');
+
+  var cardNumber = elements.create('cardNumber', {style: style});
+  cardNumber.mount('#card_number');
+
+  return cardNumber;
+}
+
+function setUpPaymentRequest(config) {
+  if (typeof config !== 'undefined') {
+    var paymentRequest = stripe.paymentRequest({
+      country: config.country,
+      currency: config.currency,
+      total: {
+        label: config.label,
+        amount: config.amount
+      },
+      requestPayerName: true,
+      requestPayerEmail: true,
+      requestShipping: config.requestShipping,
+      shippingOptions: [
+      ]
+    });
+
+    var prButton = elements.create('paymentRequestButton', {
+      paymentRequest: paymentRequest
+    });
+
+    paymentRequest.canMakePayment().then(function(result) {
```

Comment:
I'd go with this version that hides the main wrapper if there's no payment button available.

```js
    document.getElementById('stripe-payment-request').style.display = 'none';
    paymentRequest.canMakePayment().then(function(result) {
      if (result) {
        document.getElementById('stripe-payment-request').style.display = '';
        prButton.mount('#payment-request-button');
      } else {
        document.getElementById('stripe-payment-request').style.display = 'none';
      }
    });
```